ios/utils.swift

```diff
@@ -27,20 +27,18 @@ func getKeyFromKeychainById(_ id: String) throws -> SecKey {
     kSecReturnRef as String: true,
   ]
 
-  // TODO: handle not found
   var result: AnyObject?
   let status = SecItemCopyMatching(query as CFDictionary, &result)
 
-  if status != 0 {
-    let errorString = SecCopyErrorMessageString(status, nil)
-
-    throw SecureEnvironmentError.KeychainQueryError(errorString as String?)
-  }
-
-  if CFGetTypeID(result) == SecKeyGetTypeID() {
-    let key = result as! SecKey
-    return key
+  switch status {
+  case 0:
+    if CFGetTypeID(result) == SecKeyGetTypeID() {
+      let key = result as! SecKey
+      return key
+    } else {
+      throw SecureEnvironmentError.NoKeyFoundWithId(id)
```

Comment:
We should also have a JS error class?

Reply:
yeah I am working on the correct errors right now.